pkg/engine/lifecycle_test.go

```diff
@@ -1372,9 +1366,8 @@ func TestRefreshInitFailure(t *testing.T) {
 	}
 
 	//
-	// Refresh DOES fail, causing the new initialization error to appear.
+	// Refresh DOES NOT fail, though the initialization error is still persisted.
 	//
-	refreshShouldFail = true
 	p.Steps = []TestStep{{Op: Refresh, SkipPreview: true, ExpectFailure: true}}
```

Comment:
Sounds like you don't want `ExpectFailure: true`?

Reply:
I'm still making the test work.